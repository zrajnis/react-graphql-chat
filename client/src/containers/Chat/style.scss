.chat {
  display: flex;
  flex-direction: column;
  height: calc(100vh - 60px);
  margin: 30px auto;
  width: 920px;

  label {
    display: none;
  }
}

.row {
  display: flex;
  height: calc(100vh - 121px);
  width: 100%;
}
