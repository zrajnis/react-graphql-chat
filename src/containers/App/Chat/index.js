import React, { Component } from 'react'
import { graphql } from 'react-apollo'

import {ALL_MESSAGES_QUERY} from 'queries/chat'
import Chatbox from 'components/Chatbox'
import SubmitBar from 'containers/App/SubmitBar'

class Chat extends Component {
  constructor () {
    super()
    this.state = {
      from: 'anonymous',
      content: '',
    }
  }

  componentDidMount () {
    const from = window.prompt('username')
    from && this.setState({from})
  }

  render () {
    const allMessages = this.props.allMessagesQuery.allMessages || []
    return (
      <div className="chat">
        <div className="container">
          <h1>React GraphQL Chat</h1>
          {allMessages.map(message => (
            <Chatbox key={message.id} message={message} />
          ))}
        </div>
        <SubmitBar />
      </div>
    )
  }
}

export default graphql(ALL_MESSAGES_QUERY, {name: 'allMessagesQuery'})(Chat)
