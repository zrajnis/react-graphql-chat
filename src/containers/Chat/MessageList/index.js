import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

import MessageBox from 'containers/Chat/MessageList/MessageBox'

const MessageList = ({ allMessages, from }) =>
  <Fragment>
    {allMessages.map(message => (
      <MessageBox
        key={message.id}
        message={message}
        myMessage={message.from === from}
      />
    ))}
  </Fragment>

MessageList.propTypes = {
  allMessages: PropTypes.arrayOf(
    PropTypes.shape({
      content: PropTypes.string.isRequired,
      from: PropTypes.string.isRequired
    })
  ),
  from: PropTypes.string.isRequired
}

export default MessageList
