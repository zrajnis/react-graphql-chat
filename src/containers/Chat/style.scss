.chat {
  margin: 30px auto;
  width: 720px;

  .message-container {
    margin: 0;
  }

  label {
    display: none;
  }
}
