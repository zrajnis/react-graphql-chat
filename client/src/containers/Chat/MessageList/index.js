import React, { Component } from 'react'
import PropTypes from 'prop-types'

import MessageBox from 'containers/Chat/MessageList/MessageBox'
import './style.scss'

class MessageList extends Component {
  componentDidMount () {
    this.messagesEnd.scrollIntoView({ behavior: 'smooth' })
  }

  componentDidUpdate () {
    this.messagesEnd.scrollIntoView({ behavior: 'smooth' })
  }

  render () {
    const { allMessages, username } = this.props

    return (
      <ul styleName='message-list'>
        {allMessages.map(message => (
          <MessageBox
            key={message.id}
            message={message}
            myMessage={message.from === username}
          />
        ))}
        <li ref={(el) => { this.messagesEnd = el }} />
      </ul>
    )
  }
}

MessageList.propTypes = {
  allMessages: PropTypes.arrayOf(
    PropTypes.shape({
      content: PropTypes.string.isRequired,
      from: PropTypes.string.isRequired
    })
  ),
  username: PropTypes.string.isRequired
}

export default MessageList
