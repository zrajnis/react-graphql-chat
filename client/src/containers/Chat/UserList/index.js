import React from 'react'
import PropTypes from 'prop-types'

import UserBox from 'containers/Chat/UserList/UserBox'
import './style.scss'

const UserList = ({ allUsers, username }) =>
  <div styleName='user-list-wrapper'>
    <em styleName='user-list-title'>USERS ONLINE</em>
    <ul styleName='user-list'>
      {allUsers.map(user => (
        <UserBox
          key={user.id}
          myUsername={user.name === username}
          username={user.name}
        />
      ))}
    </ul>
  </div>

UserList.propTypes = {
  allUsers: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired
    })
  ),
  username: PropTypes.string.isRequired
}

export default UserList
