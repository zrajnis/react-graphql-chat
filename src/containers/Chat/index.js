import React, { Component } from 'react'
import PropTypes from 'prop-types'

import MessageBox from 'components/MessageBox'
import SubmitBar from 'components/SubmitBar'
import './style.scss'

class Chat extends Component {
  constructor (props) {
    super(props)
    this.mapMessages = this.mapMessages.bind(this)
    this.scrollToBottom = this.scrollToBottom.bind(this)
  }

  mapMessages () {
    const { allMessages, from } = this.props

    return allMessages.map(message => (
      <MessageBox
        key={message.id}
        message={message}
        myMessage={message.from === from}
      />
    ))
  }

  componentDidMount () {
    this.scrollToBottom()
  }

  componentDidUpdate () {
    this.scrollToBottom()
  }

  scrollToBottom () {
    this.messagesEnd.scrollIntoView({ behavior: 'smooth' })
  }

  render () {
    const { mapMessages } = this
    const { content, handleChange, handleSubmit } = this.props

    return (
      <div styleName='chat'>
        <div styleName='message-container'>
          {mapMessages()}
        </div>
        <SubmitBar
          buttonText='Send'
          handleChange={handleChange}
          handleSubmit={handleSubmit}
          inputVal={content}
          label='Insert a text message'
          legend='Insert a text message'
        />
        <div ref={(el) => { this.messagesEnd = el }} />
      </div>
    )
  }
}

Chat.propTypes = {
  allMessages: PropTypes.array.isRequired,
  content: PropTypes.string.isRequired,
  from: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired
}

export default Chat
