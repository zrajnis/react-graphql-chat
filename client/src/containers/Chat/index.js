import React from 'react'
import PropTypes from 'prop-types'

import MessageList from 'containers/Chat/MessageList'
import SubmitBar from 'components/SubmitBar'
import UserList from 'containers/Chat/UserList'
import './style.scss'

const Chat = ({ allMessages, allUsers, content, username, handleChange, handleSubmit }) =>
  <div styleName='chat'>
    <div styleName='row'>
      <MessageList
        allMessages={allMessages}
        username={username}
      />
      <UserList
        allUsers={allUsers}
        username={username}
      />
    </div>
    <SubmitBar
      buttonText='Send'
      handleChange={handleChange}
      handleSubmit={handleSubmit}
      inputVal={content}
      label='Insert a text message'
      legend='Insert a text message'
    />
  </div>

Chat.propTypes = {
  allMessages: PropTypes.arrayOf(
    PropTypes.shape({
      content: PropTypes.string.isRequired,
      from: PropTypes.string.isRequired,
      id: PropTypes.string.isRequired
    })
  ),
  allUsers: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired
    })
  ),
  content: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  username: PropTypes.string.isRequired
}

export default Chat
