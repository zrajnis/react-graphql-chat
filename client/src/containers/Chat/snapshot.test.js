import React from 'react'
import renderer from 'react-test-renderer'

import Chat from 'containers/Chat'

describe('Chat container snapshot', () => {
  const mockFn = jest.fn()
  const username = 'testUser'
  const allMessages = [{
    content: 'test content 123',
    from: username,
    id: '1'
  }, {
    content: 'test content 456',
    from: 'testUser1',
    id: '2'
  }, {
    content: 'test content 678',
    from: 'testUser2',
    id: '3'
  }]
  const allUsers = [{
    id: '1',
    name: 'test1'
  }, {
    id: '2',
    name: username
  }]
  const content = 'test content'
  const createNodeMock = () => ({
    scrollIntoView () {}
  })

  it('renders the chat container', () => {
    const tree = renderer.create(
      <Chat
        allMessages={allMessages}
        allUsers={allUsers}
        content={content}
        handleChange={mockFn}
        handleSubmit={mockFn}
        username={username}
      />, { createNodeMock }
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
