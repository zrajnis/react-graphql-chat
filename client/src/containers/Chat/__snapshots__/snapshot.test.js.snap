// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat container snapshot renders the chat container 1`] = `
<div
  className="chat--2BYVCiUn"
>
  <div
    className="row--vs3UeeEQ"
  >
    <ul
      className="message-list--3MC3ek3B"
    >
      <li
        className="my-message-wrapper--h93b8ATj"
      >
        <div
          className="chat-message--QnC2_lx7"
        >
          <p>
            test content 123
          </p>
        </div>
      </li>
      <li
        className="chat-message-wrapper--hp6HY0Bp"
      >
        <div
          className="chat-message--QnC2_lx7"
        >
          <span>
            testUser1
            :
          </span>
          <p>
            test content 456
          </p>
        </div>
      </li>
      <li
        className="chat-message-wrapper--hp6HY0Bp"
      >
        <div
          className="chat-message--QnC2_lx7"
        >
          <span>
            testUser2
            :
          </span>
          <p>
            test content 678
          </p>
        </div>
      </li>
      <li />
    </ul>
    <div
      className="user-list-wrapper--3Kx3AgCl"
    >
      <em
        className="user-list-title--3aw-OTIE"
      >
        USERS ONLINE
      </em>
      <ul
        className="user-list--AtQMgPsm"
      >
        <li
          className="username--ThdCkMwR"
        >
          test1
        </li>
        <li
          className="my-username--2An5F1qD"
        >
          testUser
           (You)
        </li>
      </ul>
    </div>
  </div>
  <form
    onSubmit={[MockFunction]}
  >
    <fieldset>
      <legend>
         
        Insert a text message
         
      </legend>
      <div
        className="input-container--316X5DI-"
      >
        <label
          htmlFor="textField"
        >
           
          Insert a text message
           
        </label>
        <input
          className="input--3L2xwewm"
          id="textField"
          onChange={[MockFunction]}
          type="text"
          value="test content"
        />
      </div>
      <button
        className="button--3j7htfnZ"
        id="sendBtn"
        type="submit"
      >
        Send
      </button>
      
    </fieldset>
  </form>
</div>
`;
